### ISSUE TITLE
Possible false positive: unnecessary warning for unset()

### ISSUE THREAD

**Issue by Vectorial1024** [vimeo/psalm#10425]
From the PHP manual user contributed notes: https://www.php.net/manual/en/function.unset.php

> This is probably trivial but there is no error for unsetting a non-existing variable.

In Psalm, a warning is given as long as the variable does not exist. However, this error is quite unnecessary when warning against unsetting a non-existing variable.

Sample code:

```php
<?php
unset($hello);
// Cannot find referenced variable $hello in global scope
```

The above code will actually execute with no notices, warnings, or exceptions.

**Comment by Vectorial1024** [comment]
The issue can be reproduced via https://psalm.dev/r/c0d0ef2808

Interestingly, Psalm currently can ignore the "undefined variable" "warning" when doing `isset()`, which also has the same behavior of no-error-if-undef.

**Comment by danog** [comment]
While this is technically not an issue as no warnings are emitted by PHP, reporting errors for unsets on undefined variables can be useful to find and remove unsets on variables that are actually never set anywhere (i.e. because they were removed from the codebase a long time ago, but the unsets were forgotten), so this is not a bug.

**Comment by Vectorial1024** [comment]
So this is supposed to be a code quality check, got it

@danog then I guess we can close this

**Comment by orklah** [comment]
Also it helps find bugs due to typo in variable names :)
